tensorly/decomposition/candecomp_parafac.py

```diff
@@ -231,16 +220,15 @@ def parafac(tensor, rank, n_iter_max=100, init='svd', svd='numpy_svd', tol=1e-8,
             factors[mode] = factor
 
         if tol:
+            # ||tensor - rec||^2 = ||tensor||^2 + ||rec||^2 - 2*<tensor, rec>
             # This is ||kruskal_to_tensor(factors)||^2
             factors_norm = tl.sum(tl.prod(tl.stack([tl.dot(tl.transpose(f), f) for f in factors], 0), 0))
             # mttkrp and factor for the last mode. This is equivalent to the
             # inner product <tensor, factorization>
             iprod = tl.sum(mttkrp*factor)
-            # Subtract iprod from each term to avoid loss of significance
-            rec_error = tl.sqrt(tl.abs((norm_tensor**2 - iprod) + (factors_norm - iprod))) / norm_tensor
+            rec_error = tl.sqrt(tl.abs(norm_tensor**2 + factors_norm - 2*iprod)) / norm_tensor
```

Comment:
Why did you undo this? I seem to remember it making a difference in some cases.

Reply:
I just grouped the 2*iprod to be easier to read as the direct expansion of the squared norm. 
Why would there be a difference?